test/integration/puppetlabs/services/jruby/class_info_test.clj

```diff
@@ -0,0 +1,229 @@
+(ns puppetlabs.services.jruby.class-info-test
+  (:require [clojure.test :refer :all]
+            [puppetlabs.kitchensink.core :as ks]
+            [puppetlabs.services.jruby.jruby-testutils :as jruby-testutils]
+            [puppetlabs.services.jruby.jruby-puppet-internal :as jruby-internal]
+            [me.raynes.fs :as fs]
+            [cheshire.core :as cheshire])
+  (:import (java.util.concurrent LinkedBlockingDeque)))
+
+(deftest ^:integration class-info-test
+  (testing "class info properly enumerated for"
+    (let [pool (LinkedBlockingDeque. 1)
```

Comment:
I think it would be worth pullilng the functions out into their own `defn`s outside of the `deftest` since there's a large number of them.

I'm happy to be overruled on this point, though, if others disagree.